### ISSUE TITLE
False positive unused dependencies in default Expo template

### ISSUE THREAD

**Issue by ProchaLu** [depcheck/depcheck#918]
### Bug Description

When generating a new Expo project with the default template (`npx create-expo-app@latest`), `depcheck` incorrectly flags several dependencies as unused. However, these dependencies are included intentionally and are likely utilized during the build process rather than through direct imports.

```json
Unused dependencies
* expo-system-ui
* react-dom
* react-native-gesture-handler
* react-native-web
Unused devDependencies
* @types/jest
* typescript
```

You can reproduce the issue by cloning this repository and running `depcheck`:

```bash
git clone https://github.com/ProchaLu/expo-depcheck-unused-dependencies.git
cd expo-depcheck-unused-dependencies
pnpm install # or npm install / yarn install
npx depcheck --json
```

Where the dependency is listed in package.json:

package.json from the [Expo default template example](https://github.com/expo/expo/blob/main/templates/expo-template-default/package.json)

```
{
  "name": "expo-depcheck-unused-dependencies",
  "main": "expo-router/entry",
  "version": "1.0.0",
  "scripts": {
    "start": "expo start",
    "reset-project": "node ./scripts/reset-project.js",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "test": "jest --watchAll",
    "lint": "expo lint"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "@expo/vector-icons": "^14.0.2",
    "@react-navigation/bottom-tabs": "^7.2.0",
    "@react-navigation/native": "^7.0.14",
    "expo": "~52.0.37",
    "expo-blur": "~14.0.3",
    "expo-constants": "~17.0.7",
    "expo-font": "~13.0.4",
    "expo-haptics": "~14.0.1",
    "expo-linking": "~7.0.5",
    "expo-router": "~4.0.17",
    "expo-splash-screen": "~0.29.22",
    "expo-status-bar": "~2.0.1",
    "expo-symbols": "~0.2.2",
    "expo-system-ui": "~4.0.8",
    "expo-web-browser": "~14.0.2",
    "react": "18.3.1",
    "react-dom": "18.3.1",
    "react-native": "0.76.7",
    "react-native-gesture-handler": "~2.20.2",
    "react-native-reanimated": "~3.16.1",
    "react-native-safe-area-context": "4.12.0",
    "react-native-screens": "~4.4.0",
    "react-native-web": "~0.19.13",
    "react-native-webview": "13.12.5"
  },
  "devDependencies": {
    "@babel/core": "^7.25.2",
    "@types/jest": "^29.5.12",
    "@types/react": "~18.3.12",
    "@types/react-test-renderer": "^18.3.0",
    "jest": "^29.2.1",
    "jest-expo": "~52.0.4",
    "react-test-renderer": "18.3.1",
    "typescript": "^5.3.3"
  },
  "private": true
}
```

### Versions

- `node -v`: v22.13.1
- `npm -v`: 10.9.2
- `pnpm -v`: 10.3.0
- `depcheck --version`: 1.4.7

### Extra info

This issue occurs in a newly generated Expo project without any changes. These dependencies are likely required at build time rather than in direct imports.

An issue was reported in 2021 but was closed as stale: [#673](https://github.com/depcheck/depcheck/issues/673).

#### `depcheck --json`

```json
{
  "dependencies": [
    "expo-system-ui",
    "react-dom",
    "react-native-gesture-handler",
    "react-native-web"
  ],
  "devDependencies": ["@types/jest", "typescript"],
  "using": {
    "expo": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/package.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/package.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/tsconfig.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HapticTab.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/TabBarBackground.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/+not-found.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/_layout.tsx"
    ],
    "jest": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/package.json"
    ],
    "jest-expo": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/package.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/package.json"
    ],
    "react": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/package.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/package.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/tsconfig.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/hooks/useColorScheme.ts",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/hooks/useColorScheme.web.ts",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/Collapsible.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HapticTab.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HelloWave.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ParallaxScrollView.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ThemedText.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ThemedView.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/TabBarBackground.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/__tests__/ThemedText-test.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/+not-found.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/explore.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/index.tsx"
    ],
    "react-native": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/package.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/package.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/tsconfig.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/hooks/useColorScheme.ts",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/hooks/useColorScheme.web.ts",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/Collapsible.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HapticTab.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HelloWave.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ParallaxScrollView.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ThemedText.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ThemedView.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/TabBarBackground.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/+not-found.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/explore.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/index.tsx"
    ],
    "react-native-webview": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/package.json",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/tsconfig.json"
    ],
    "@types/react": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/hooks/useColorScheme.ts",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/hooks/useColorScheme.web.ts",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/Collapsible.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HelloWave.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ParallaxScrollView.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ThemedText.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ThemedView.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/TabBarBackground.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/__tests__/ThemedText-test.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/+not-found.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/explore.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/index.tsx"
    ],
    "expo-router": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/+not-found.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/_layout.tsx"
    ],
    "expo-web-browser": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx"
    ],
    "expo-constants": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/+not-found.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/_layout.tsx"
    ],
    "expo-linking": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/+not-found.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/_layout.tsx"
    ],
    "react-native-reanimated": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HelloWave.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ParallaxScrollView.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/+not-found.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/_layout.tsx"
    ],
    "react-native-safe-area-context": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HapticTab.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/TabBarBackground.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/+not-found.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/_layout.tsx"
    ],
    "react-native-screens": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ExternalLink.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HapticTab.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/TabBarBackground.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/+not-found.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/(tabs)/_layout.tsx"
    ],
    "@react-navigation/bottom-tabs": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HapticTab.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/TabBarBackground.ios.tsx"
    ],
    "@react-navigation/elements": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HapticTab.tsx"
    ],
    "expo-haptics": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HapticTab.tsx"
    ],
    "@react-navigation/native": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HapticTab.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/TabBarBackground.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx"
    ],
    "@babel/core": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/HelloWave.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ParallaxScrollView.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx"
    ],
    "expo-symbols": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.ios.tsx",
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.tsx"
    ],
    "@expo/vector-icons": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/IconSymbol.tsx"
    ],
    "expo-blur": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/ui/TabBarBackground.ios.tsx"
    ],
    "react-test-renderer": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/__tests__/ThemedText-test.tsx"
    ],
    "@types/react-test-renderer": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/components/__tests__/ThemedText-test.tsx"
    ],
    "expo-font": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx"
    ],
    "expo-splash-screen": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx"
    ],
    "expo-status-bar": [
      "/Users/lukas/Documents/projects/expo-depcheck-unused-dependencies/app/_layout.tsx"
    ]
  },
  "invalidFiles": {},
  "invalidDirs": {}
}
```
